packages/vue-component-library/src/lib-components/SearchInput.vue

```diff
@@ -1,293 +1,127 @@
-    // console.log('in search input watch', newVal)
-    searchInputModelValue.value = newVal
-  },
-  { immediate: true }
-)
-
-onBeforeUnmount(() => {
-  window.document.removeEventListener('keydown', onDocumentKeydown)
-})
+// Data
+const inputRef = ref<HTMLInputElement | null>(null)
 
+// Methods
 function clear() {
-  // console.log('in clear')
-  searchInputModelValue.value = ''
   emit('update:modelValue', '')
   emit('clear')
 }
+function onInput(event: Event) {
+  const target = event.target as HTMLInputElement
 
-function onInput(e) {
-  searchInputModelValue.value = e.target.value
-  emit('update:modelValue', e.target.value)
+  emit('update:modelValue', target.value)
 }
-
-function onKeydown(e) {
-  if (e.key === 'Escape') {
+function onKeydown(event: KeyboardEvent) {
+  if (event.key === 'Escape') {
+    event.preventDefault()
     if (props.clearOnEsc)
       clear()
-    if (props.blurOnEsc) {
-      const el = inputRef.value
-      if (el)
-        el.blur()
-    }
+    if (props.blurOnEsc)
+      inputRef.value?.blur()
   }
 }
-
-function onDocumentKeydown(e) {
+function onDocumentKeydown(event: KeyboardEvent) {
   if (
-    e.key === props.shortcutKey
-    && e.target !== inputRef.value
-    && window.document.activeElement !== inputRef.value
-    && !(e.target instanceof HTMLInputElement)
-    && !(e.target instanceof HTMLSelectElement)
-    && !(e.target instanceof HTMLTextAreaElement)
-  ) {
-    e.preventDefault()
-    const allVisibleSearchInputs = [].slice
-      .call(
-        document.querySelectorAll(
-          '[data-search-input="true"]:not([data-shortcut-enabled="false"])'
+    event.key === props.shortcutKey
+        && event.target !== inputRef.value
+        && document.activeElement !== inputRef.value
+        && !['INPUT', 'SELECT', 'TEXTAREA'].includes(
+          (event.target as Element).tagName
         )
+  ) {
+    event.preventDefault()
+    const visibleInputs = Array.from(
+      document.querySelectorAll<HTMLInputElement>(
+        '[data-search-input="true"]:not([data-shortcut-enabled="false"])'
       )
-      .filter((el) => {
-        return !!(
-          el.offsetWidth
-          || el.offsetHeight
-          || el.getClientRects().length
-        )
-      })
+    ).filter(
+      el =>
+        el.offsetWidth || el.offsetHeight || el.getClientRects().length
+    )
     const elToFocus
-      = allVisibleSearchInputs.length > 1
-        ? allVisibleSearchInputs[0]
-        : inputRef.value
-
+            = visibleInputs.length > 1 ? visibleInputs[0] : inputRef.value
     if (elToFocus) {
       elToFocus.focus()
       if (props.selectOnFocus)
         elToFocus.select()
     }
   }
 }
+
+// Computeds
+const isClearIconShown = computed(() => {
+  console.log('props.clearIcon', props.clearIcon)
+  console.log('props.modelValue', props.modelValue)
+  return props.clearIcon && !!props.modelValue && props.modelValue.length > 0
+})
+const classes = computed(() => [
+  'search-input-wrapper',
```

Comment:
since you are using the class as search-input-wrapper, you need to use this in spec . This will fix cypress error on this story